### ISSUE TITLE
The boundary value of judging whether two person is the same

### ISSUE THREAD

**Issue by chikaishiro** [VisualComputingInstitute/triplet-reid#51]
Hello ! 
I'm now using the pre-trained model of Market1501.
My purpose is to judge whether a person can be found in a given dataset.
So I'm looking for the boundary value of distance to judge.
What's the recommended value of distance?

**Comment by Pandoro** [comment]
This goes towards open set ReID, something we did not look into. But there
is no such value for all models and setups. It will change depending on
many factors, likely even from one training to the next. To approximate
this you will have to make two histograms of positive and negative
distances and pick a value depending on whether false postives or false
negatives are a bigger problem for your scenario.

The question is on what annotated data you can do this. If you use the
market test set and then assume this value generalized to another dataset,
this could possibly go wrong. But in any case, pick a new value after every
training.

**Comment by lucasb-eyer** [comment]
To make it perfectly clear: this value depends 1) on the model 2) on the model's weights and 3) on the images. You need to use some data-driven approach to find it, and re-compute it every time one of the three changes!

**Comment by voqtuyen** [comment]
@lucasb-eyer. If we use hard-margin of 0.5 for example, and we judge the similarity of two persons by thresholding 0.5 if the test set is similar to train set, right?

**Comment by Pandoro** [comment]
No, this does not need to be the case for several reasons:

1. The distance between all positives could go up to, for example 2.3, the
hard margin of 0.5 in that case would only mean that all negatives are
farther away than 2.3+0.5.

2. Even if your test set is similar. In our setup the train and test set
never contain the same person. Empirically, they network produces a by far
noisier embedding for persons it has not seen, thus likely the actual
margin between positiv and negative distances will be different on the test
set and there will not be a single threshold that separates all persons in
your test set.

You will typically have to make a histogram where you compare all positives
to all negative distances in an unseen validation set, or if you don't care
about doing everything correctly on your test set. This threshold will then
trade off the number of false negative matches vs false positiv matches,
where you can impose a personalized weight for the two, to determine the
threshold that fits your application.
